pkg/stanza/operator/input/windows/event.go

```diff
@@ -18,54 +18,64 @@ type Event struct {
 	handle uintptr
 }
 
+// NewEvent will create a new event from an event handle.
+func NewEvent(handle uintptr) Event {
+	return Event{
+		handle: handle,
+	}
+}
+
 // RenderSimple will render the event as EventXML without formatted info.
-func (e *Event) RenderSimple(buffer Buffer) (EventXML, error) {
+func (e *Event) RenderSimple(buffer Buffer) (*EventXML, error) {
 	if e.handle == 0 {
-		return EventXML{}, fmt.Errorf("event handle does not exist")
+		return nil, fmt.Errorf("event handle does not exist")
 	}
 
 	bufferUsed, err := evtRender(0, e.handle, EvtRenderEventXML, buffer.SizeBytes(), buffer.FirstByte())
-	if errors.Is(err, ErrorInsufficientBuffer) {
-		buffer.UpdateSizeBytes(*bufferUsed)
-		return e.RenderSimple(buffer)
-	}
-
 	if err != nil {
-		return EventXML{}, fmt.Errorf("syscall to 'EvtRender' failed: %w", err)
+		if errors.Is(err, ErrorInsufficientBuffer) {
+			// If the bufferUsed is 0 return an error as we don't want to make a recursive call with no buffer
+			if *bufferUsed == 0 {
```

Comment:
Nice, this shouldn't happen per API description, but, LGTM.

Reply:
Yeah, I noticed we were doing it in both `RenderRaw` (now removed) and `RenderFormatted`, so just put it here too. If you're confident it's not an issue then I'm happy to remove it all around.

Reply:
The API will be really broken if it is zero, I'm comfortable removing it.

Reply:
Ok, removed it